ui/raidboss/timeline.ts

```diff
@@ -63,8 +68,72 @@ const activeText = {
   ko: '시전중:',
 };
 
+type Replacement = {
+  locale: string;
+  replaceSync: { [key: string]: string };
+  replaceText: { [key: string]: string };
+};
+
+type Style = {
+  style: { [key: string]: string };
+  regex: RegExp;
+}
+
+type Event = {
+  id: number;
+  time: number;
+  name: string;
+  text: string;
+  activeTime?: number;
+  lineNumber?: number;
+  duration?: number;
+  sortKey?: number;
+  isDur?: boolean;
+  style?: { [key: string]: string };
+}
+
+type Error = {
+  lineNumber?: number;
+  line?: string;
+  error: string;
+}
+
+type Sync = {
+  id: number;
+  origRegexStr: string;
+  regex: RegExp;
+  start: number;
+  end: number;
+  time: number;
+  lineNumber: number;
+  jump?: number;
+}
+
+type ParsedText = {
+  type: 'info' | 'alert' | 'alarm' | 'tts';
+  secondsBefore?: number;
+  text: string;
+} | {
+  type: 'trigger';
+  secondsBefore?: number;
+  text?: string;
+  matches: RegExpExecArray | null;
+  trigger: Trigger;
+}
+
+type Text = ParsedText & { time: number };
+
+interface RaidbossOptions extends BaseOptions {
+  TimelineLanguage: Lang;
+  MaxNumberOfTimerBars: number;
+  BarExpiresSoonSeconds: number;
+  ShowTimerBarsAtSeconds: number;
+  KeepExpiredTimerBarsForSeconds: number;
+  PerTriggerAutoConfig: { [triggerId: string]: TriggerAutoConfig };
+}
```

Comment:
Sorry for the conflict but I just checked in the top level raidboss.ts conversion, which adds a RaidbossOptions type to `raidboss_options.ts`.  Can this get merged into there?